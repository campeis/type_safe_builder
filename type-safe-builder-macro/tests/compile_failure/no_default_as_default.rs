use type_safe_builder_macro::Builder;

struct WithNoDefault {
    field: i64,
}
#[derive(Builder)]
struct StructWithFieldWithNoDefault {
    #[build_default]
    field: WithNoDefault,
}

fn main() {}
