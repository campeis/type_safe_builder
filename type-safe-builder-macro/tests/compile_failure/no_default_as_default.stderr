error[E0277]: the trait bound `WithNoDefault: Default` is not satisfied
 --> tests/compile_failure/no_default_as_default.rs:6:10
  |
6 | #[derive(Builder)]
  |          ^^^^^^^ the trait `Default` is not implemented for `WithNoDefault`
  |
note: required by a bound in `Option::<T>::unwrap_or_default`
 --> $RUST/core/src/option.rs
  |
  |     pub fn unwrap_or_default(self) -> T
  |            ----------------- required by a bound in this associated function
  |     where
  |         T: Default,
  |            ^^^^^^^ required by this bound in `Option::<T>::unwrap_or_default`
  = note: this error originates in the derive macro `Builder` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `WithNoDefault` with `#[derive(Default)]`
  |
3    + #[derive(Default)]
4    | struct WithNoDefault {
     |
